### PR TITLE
feat: added exadsadserver.md for PR purposes

### PR DESCRIPTION
**Pull request by giuseppe-exads**
Please check mainly the first part of the document (exadsadserver.md). 
I've left some question marks as place-holder near to some attributes. Visit https://docs.prebid.org/dev-docs/bidder-adaptor.html and search for "Create a docs pull request against [prebid.github.io](http://prebid.github.io/)").

### CODE REVIEW

File: dev-docs/bidders/exadsadserver.md [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+tcfeu_supported: true?
+usp_supported: false?
+media_types: banner, video, native
+gpp_sids: ?
+schain_supported: ?
+dchain_supported: ?
+safeframes_ok: ?
+deals_supported: ?
+floors_supported: ?
+ortb_blocking_supported: ?
```
Comment (broucz) [inline dev-docs/bidders/exadsadserver.md]: @giuseppe-exads Could you provide us with a link to a place that document these fields? Thanks!

File: dev-docs/bidders/exadsadserver.md [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+---
+
+### Note
+
+Module that connects to EXADS’ bidder for bids.
+
```
Comment (broucz) [inline dev-docs/bidders/exadsadserver.md]: @giuseppe-exads I wouldn't keep this section, it bring no useful information.

Alternatively, we could have a top section for a menu as our documentation is quite long, and add the information to contact us in cache of an integration issue.

File: dev-docs/bidders/exadsadserver.md [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+### Configuration
```
Comment (broucz) [inline dev-docs/bidders/exadsadserver.md]: @giuseppe-exads Before going into various configuration example, I would first add a simple and minimal example, something like you've done for the "RTB Banner 2.4". Then, I would add a section for each format with details of their specificity and configuration, from the simpler to the more complex one.

File: dev-docs/bidders/exadsadserver.md [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+    },
+});
+```
+
+#### RTB Banner 2.4
+
+{: .table .table-bordered .table-striped }
+| Name          | Scope    | Description           | Example   | Type      |
+|---------------|----------|-----------------------|-----------|-----------|
+| `zoneID`      | required | You can get it from the endpoint created after configuring the zones          | `12345`    | `'integer'` |
```
Comment (broucz) [inline dev-docs/bidders/exadsadserver.md]: @giuseppe-exads Having both " \` " and " ' " for the type is too much. Use " \` " only.

File: dev-docs/bidders/exadsadserver.md [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+| `fid`      | required | You can get it from the endpoint created after configuring the zones          | `'829a896f011475d50da0d82cfdd1af8d9cdb07ff'`    | `'string'` |
+| `partner`      | required | Currently we support rtb 2.4 (“ortb_2_4”) only           | `'ortb_2_4'`    | `'string'` |
+| `siteId`      | recommended |  Unique Site ID           | `'123'`    | `'string'` |
+| `siteName`      |  |  Site name            | `'test.com'`    | `'string'` |
+| `banner.sizes`      | required |  [width, height]          | `[145,256]`    | `'integer array'` |
+| `userIp`      | required |  IP address of the user, ipv4 or ipv6          | `'0.0.0.0'`    | `'string'` |
+| `userId`      | *required |  Unique user ID (string). *If you cannot generate a user ID, you can leave it empty (""). The request will get a response as long as “user” object is included in the request          | `''`    | `'string'` |
+| `country`      |  |  country ISO3          | `'IRL'`    | `'string'` |
```
Comment (broucz) [inline dev-docs/bidders/exadsadserver.md]: @giuseppe-exads Please specify then it's optional by writing it in the scope section.

File: dev-docs/bidders/exadsadserver.md [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+                ]
+            }
+        ],
+        "adrender": ...
+    }
+}
+```
+For more information navigate to https://docs.prebid.org/dev-docs/bidder-adaptor.html.
+
+# Tools and suggestions
```
Comment (broucz) [inline dev-docs/bidders/exadsadserver.md]: @giuseppe-exads I wouldn't keep this "Tools and suggestions" section. It's a bit unrelated to the purpose of this doc.

File: dev-docs/bidders/exadsadserver.md [hunk 7]
```diff
@@ -0,0 +1,487 @@
...
+                    dsa: {
+                        dsarequired: 3, 
+                        pubrender: 0,
+                        datatopub: 2
+                    },
+                    endpoint: 'https://your-ad-network.com/rtb.php'
+                }
+            }]
+        }];
+```
```
Comment (broucz) [inline dev-docs/bidders/exadsadserver.md]: @giuseppe-exads In these snippets, we should probably use standard Javascript formatting and indentation (c.f., Prettier defaults).

I would also add `// optional` to the parameters that aren't required like in:
https://github.com/prebid/prebid.github.io/blob/master/dev-docs/bidders/openx.md#adunit-format-for-video

File: dev-docs/bidders/exadsadserver.md (hunk 1 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: @broucz [The link is that mentioned above:  https://docs.prebid.org/dev-docs/bidder-adaptor.html.](https://github.com/EXADS/prebid.github.io/pull/2#issue-2175894774)

File: dev-docs/bidders/exadsadserver.md (hunk 2 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: @broucz Note section removed

File: dev-docs/bidders/exadsadserver.md (hunk 4 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: done

File: dev-docs/bidders/exadsadserver.md (hunk 5 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: done

File: dev-docs/bidders/exadsadserver.md (hunk 6 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: Removed section.

File: dev-docs/bidders/exadsadserver.md (hunk 7 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: Snippet code formatted using https://prettier.io/playground/.
Added `// optional` where needed.

File: dev-docs/bidders/exadsadserver.md [hunk 8]
```diff
@@ -0,0 +1,487 @@
...
+# DSA Transparency
+All DSA information, returned by the ad server, can be found into the **meta** tag of the response. As:
+```
+"meta": {
+    "dsa": {
+        "behalf": "...",
+        "paid": "...",
+        "transparency": [
+            {
+                "params": [
```
Comment (kristijanexads) [inline dev-docs/bidders/exadsadserver.md:466]: just out of curiosity, does this support `params` and `dsaparams`? Because officially it should be `dsaparams` but we still return `params` because some of the clients didn't want to change it until standard becomes fully matured. Once we are "allowed" to return it as `dsaparams` it would be nice if this would work without any changes done on the prebid.js.
Also, there is a field `domain` that we don't return for now but we might at some point. 
Basically to support what is there now:
https://github.com/InteractiveAdvertisingBureau/openrtb/blob/main/extensions/community_extensions/dsa_transparency.md#sample-openrtb-26-bid-response-with-dsa-transparency

File: dev-docs/bidders/exadsadserver.md (hunk 6 above)
Comment (kristijanexads) [reply dev-docs/bidders/exadsadserver.md]: Agree. We could maybe have url that would take them to our documentation where they could find this extra info, but not more than that.

File: dev-docs/bidders/exadsadserver.md (hunk 1 above)
Comment (kristijanexads) [reply dev-docs/bidders/exadsadserver.md]: I would remove those where we know that default matches what we have, and for the other we should put correct values:

```
pbs: false,
gvl_id: 1084, but confirm with Barry/Tiago if everyone uses EXADS vendor id or they apply for their own,
tcfeu_supported: true,
usp_supported: false,
gpp_sids: tcfeu,
schain_supported: false,
dchain_supported: false,
safeframes_ok: false (have no idea tbh, but since we never tried I would go with false),
deals_supported: false,
floors_supported: true,
ortb_blocking_supported: true - not yet live but I'm pretty sure it will go live before Prebid team gets hands on this
```

**Review by kristijanexads — CHANGES_REQUESTED** [review]
small changes

File: dev-docs/bidders/exadsadserver.md (hunk 1 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: Changed all except glv_id for which I'm waiting feedback by @tfoliveira and Barry about gvl_id into slack.

File: dev-docs/bidders/exadsadserver.md (hunk 8 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md:466]: Relating `params`, it is not a problem because the adapter takes our response reading `ext.dsa` and putting its content into meta.dsa.
Also `transparency.domain` is returned into `ext.dsa` so it is transparent to the adapter as well, similarly as for `params`.

The only thing to do will be change the documentation when we'll manage both of attributes.

File: dev-docs/bidders/exadsadserver.md (hunk 3 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: @broucz In the beginning, before the configuration section, I've just added one very minimal snipped code without optional parameters and a brief description, in order to give a first idea about how to use the adapter.

After that, I've nearly maintained the configuration section as now because it is across over all ad formats. The only thing that I did is adding two sub paragraphs, generic and video settings, to make it more clear the content.

Anyway we can continue improving it but consider, as you know, that It is not the end user manual.

File: dev-docs/bidders/exadsadserver.md (hunk 1 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: @tfoliveira has confirmed that glv_id is 1084.

File: dev-docs/bidders/exadsadserver.md [hunk 9]
```diff
@@ -0,0 +1,475 @@
+---
+layout: bidder
+title: EXADS
+description: EXADS Bidder Adapter
+pbjs: true
+biddercode: exadsadserver
+gvl_id: 1084 - waiting Tiago/Barry
```
Comment (kristijanexads) [inline dev-docs/bidders/exadsadserver.md]: maybe better not mention names in PRs

**Review by kristijanexads — APPROVED** [review]

File: dev-docs/bidders/exadsadserver.md (hunk 9 above)
Comment (broucz) [reply dev-docs/bidders/exadsadserver.md]: @giuseppe-exads ☝️

**Review by broucz — APPROVED** [review]

File: dev-docs/bidders/exadsadserver.md (hunk 9 above)
Comment (giuseppe-exads) [reply dev-docs/bidders/exadsadserver.md]: done
